Docs/User Guide.md

```diff
@@ -0,0 +1,180 @@
+
+- Custom Device Project
+- Support Project
+
+### Custom Device Project
+
+The `Source/Custom Device/<MyName> Custom Device.lvproj` primarily exists for historical reasons. This project contains the code and build specs required for interacting with VeriStand. It builds the `.llb` files that VeriStand requires for execution.
+
+This project has the same structure as a project generated from the wizard, but contains minimal logic for performing work in the custom device. Most of the code is a set of wrappers that call code implemented in packed project libraries built from the support project. These libraries can be found in the `Includes` virtual folder in the project.
+
+### Support Project
+
+The `Source/Custom Device Support/<MyName> Support.lvproj` is the project where almost all logic for the custom device's behavior will exist. This project contains the logic for interacting with hardware, encoding/decoding bus messages, System Explorer pages, and database management. It builds packed project libraries (PPLs) in the form of `Engine.lvlibp` and `System Explorer.lvlibp` files, which support hierarchical file structures. These PPLs enable the custom device author to utilize classes and interfaces to implement extensible architectures. This strategy also promotes consistency of experience across custom devices.
+
+### Execution Unit
+
+The fundamental piece of the engine for custom devices using this template is the **Execution Unit**. The **Execution Unit** is a LabVIEW _Interface_ that a custom device author must implement in order to use this template. Using this interface enables the custom device author to focus almost entirely on only the code pieces necessary for implementing a specific protocol and communicating with a specific hardware device without having to think much about how it fits into VeriStand.
+
+The **Execution Unit** interface and the process for implementing it are discussed throughout this guide. The methods for this interface are shown below.
+
+![ExecutionUnit](Resources/ExecutionUnit.png)
+
+## Developing a New Custom Device
+
+### Scripting API
+
+All code interacting with the system definition through the LabVIEW Custom Device API, or the .NET API directly, should go through VIs contained in `<MyName> Scripting.lvlib`. This library is the single source of truth for defining anything existing in the system definition XML. The exception to this rule is if the VeriStand API already provides a VI for doing exactly what is required, i.e. `NI VeriStand - Set Item Description.vi`.
+
+This layer enables users to access the system definition from within System Explorer and directly through LabVIEW, while going through a single code path. If any code is touching the system definition through a VI not contained in this library, that functionality should be moved into this API.
```

Comment:
~If any code is touching the system definition through a VI not contained in this library, that functionality should be moved into this API.~

and then move 
>This layer enables users to access the system definition from within System Explorer and directly through LabVIEW, while going through a single code path.

into paragraph above before "The exception..."